### PR TITLE
✨(api): Adding mock authentication api

### PR DESCRIPTION
**Pull request by MoeMoola3**
Adding the Mock API. Returns hardcoded data to get the API up and running.

### CODE REVIEW

**Review by tshegomotlatle — APPROVED** [review]
Looks good to me, doesn't break the pipeline and passes all tests provided by the CI/CD team

File: libs/api/authentication/api/feature/src/lib/api-authentication-api-feature.controller.specs.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppController } from './api-authentication-api-feature.controller';
```
[the remark this answers is not in the document]
Comment (arneschreuder) [reply libs/api/authentication/api/feature/src/lib/api-authentication-api-feature.controller.specs.ts:2]: GraphQL using nest technically has a controller, but it is hidden underneath the plumbing of the @nestjs/graphql package.

File: libs/api/authentication/api/feature/src/lib/api-authentication-api-feature.controller.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Controller, Get } from '@nestjs/common';
```
Comment (arneschreuder) [inline libs/api/authentication/api/feature/src/lib/api-authentication-api-feature.controller.ts:1]: Please remove this file. There are no controllers in the API.

File: libs/api/authentication/api/feature/src/lib/main.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { NestFactory } from '@nestjs/core';
```
Comment (arneschreuder) [inline libs/api/authentication/api/feature/src/lib/main.ts:1]: No. Please remove. No main files in the libs. The executable that bundle the app is in `apps/api`.

File: libs/api/authentication/api/feature/src/lib/users/users.entity.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Field, Int, ObjectType } from "@nestjs/graphql";
+
+@ObjectType()
+export class User{
+    @Field(type => Int)
```
[the remark this answers is not in the document]
Comment (arneschreuder) [reply libs/api/authentication/api/feature/src/lib/users/users.entity.ts:5]: Or documentation for database.

File: libs/api/authentication/api/feature/src/lib/users/users.module.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { Module } from '@nestjs/common';
+import { UsersService } from './users.service';
+import { UsersResolver } from './users.resolver';
+
+@Module({
+  providers: [UsersService, UsersResolver]
```
Comment (arneschreuder) [inline libs/api/authentication/api/feature/src/lib/users/users.module.ts:6]: Services should move to service layer in `libs/api/authentication/service/feature`. Then import using `@graduates/api/authentication/service/feature`

File: libs/api/authentication/api/feature/src/lib/users/users.service.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@nestjs/common';
```
Comment (arneschreuder) [inline libs/api/authentication/api/feature/src/lib/users/users.service.ts:1]: See other comments. Services should be in service layer.

**Review by arneschreuder — CHANGES_REQUESTED** [review]
Few small changes, if that is fixed, then please go ahead and merge.

**Review by tshegomotlatle — APPROVED** [review]
All the changes mentioned by Arne seem to be done. I'll authorize the merge to the develop branch

**Pull request merged**